### PR TITLE
Adding canVariables fetcher

### PR DESCRIPTION
**Pull request by waldemarnt**
Hey folks @bltavares and I just created a fetcher that will search and fetch CAN variables from telemetry API and normalize them into a object.

### CODE REVIEW

File: fetcher/canVariablesFetcher.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import config from '../config';
+import rp from 'request-promise';
```
Comment (bltavares) [inline fetcher/canVariablesFetcher.js]: I think it is safe to remove this import, as we are injecting the client.

File: test/fetcher/canVariablesFetcher_spec.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import CanVariablesFetcher from '../../fetcher/canVariablesFetcher.js';
+
+describe('CanVariablesFetcher', () => {
+  it('fetches can variables data by equipment id', (done) => {
+
+    let expetecRequestReponse = {
```
Comment (bltavares) [inline test/fetcher/canVariablesFetcher_spec.js]: Typo: `expectedRequestResponse`

**Comment by glauco** [comment]
LGTM! :+1:

File: fetcher/canVariablesFetcher.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        data.meta.aggregations.equip_agg.forEach((data, index) => {
+          canVariables[data.key] = {};
+          data.spn_ag.forEach((aggData, index) => {
+            canVariables[data.key][aggData.key] = aggData.spn_latest_ag[0].value;
+          });
+        });
+        return canVariables;
+      })
+      .catch(function (err) {
+        console.log(err.error);
```
Comment (jeffhsta) [inline fetcher/canVariablesFetcher.js]: What will happen it the software get in this catch?

The Facade will show an error?
HTTP status code 500?

File: config.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  TELEMETRY_API_URL: process.env.TELEMETRY_API_URL || 'https://agco-fuse-trackers-sandbox.herokuapp.com'
```
Comment (jeffhsta) [inline config.js]: :+1:

File: test/server_spec.js [hunk 5]
```diff
@@ -76,20 +80,82 @@ describe('EquipmentController', () => {
...
+                    ]
+                  }
+                ]
+              }
+            ]
+          }
+        }
+      }
+
+      let mockedSearchUri = 'https://agco-fuse-trackers-sandbox.herokuapp.com/trackingData/search?include=trackingPoint&links.canVariable.name=ENGINE_HOURS,ENGINE_SPEED,DRIVING_DIRECTION&aggregations=equip_agg&equip_agg.property=links.trackingPoint.equipment.id&equip_agg.aggregations=spn_ag%2Ctp_latest_ag&spn_ag.property=links.canVariable.name&spn_ag.aggregations=spn_latest_ag&spn_latest_ag.type=top_hits&spn_latest_ag.sort=-links.trackingPoint.timeOfOccurrence&spn_latest_ag.limit=1&spn_latest_ag.include=canVariable%2CcanVariable.standardUnit&tp_latest_ag.type=top_hits&tp_latest_ag.sort=-links.trackingPoint.timeOfOccurrence&tp_latest_ag.limit=1&tp_latest_ag.fields=links.trackingPoint&tp_latest_ag.include=trackingPoint&links.trackingPoint.equipment.id=a-equipment-id-1,a-equipment-id-2';
```
Comment (jeffhsta) [inline test/server_spec.js]: could this be defined in describe scope?

File: test/server_spec.js [hunk 6]
```diff
@@ -76,20 +80,82 @@ describe('EquipmentController', () => {
       };
     });
 
-    it('request telemetry api with same authorization header', (done) => {
-      respondWithSuccess(httpClient(telemetryRequest), {
-        equipment: [],
-        links: {}
-      });
+    it('request telemetry api and receiving equipments', (done) => {
+      telemetryRequest = {
```
Comment (jeffhsta) [inline test/server_spec.js]: make sense put this telemetryRequest explicit here or define this in a describe scope that will be used for other tests/specifications?

Comment (waldemarnt) [inline test/server_spec.js]: @jeffhsta our idea was to isolate this test scenarios. Does not should be changed thought the tests.

Comment (waldemarnt) [inline test/server_spec.js]: Let's extract this mocks to fixtures? does make sense?

**Comment by bltavares** [comment]
This branch has been merged manually.

**Comment by waldemarnt** [comment]
:+1:  @bltavares

**Pull request closed without merging**